watsononlinestore/watson_online_store.py

```diff
@@ -83,6 +85,46 @@ def handle_db_lookup(self):
             print("DB.\n context:{}\n".format(self.context))
         return False
 
+    def handle_lookupAndAddEmail(self):
+        """ Verify email is not in DB and add to DB.
+        """
+        email_addr = str(self.context['email'])
+        if "mailto" in email_addr:
+            email_addr = self.cleanup_email(email_addr)
+        if DEBUG:
+            print("DB.\n email_addr:{}\n".format(email_addr))
+        existing_user = self.cloudant_online_store.find_customer(email_addr)
+        if DEBUG:
+            print("DB.\n existing_user:{}\n".format(existing_user))
+        if existing_user:
+            #return some error
+            return False
+        
+        # start to create the customer, add to DB when complete
+        self.customer = OnlineStoreCustomer(email=email_addr)
+
+        return True
+
+    def handle_AddName(self):
+        """ Add User Name to existing OnlineStoreCustomer
+        """
+        full_name = str(self.context['full_name'])
+        first, last = full_name.split()
+        self.customer.first_name = first
+        self.customer.last_name = last 
+
+        # TODO continue to ask for favorites and remove next lines, maybe,
+        #  but not for now.
+        self.customer.logged_in = True
+        self.context['logged_in'] = False
+        #logged_in = {'logged_in': True}
+        self.context = self.context_merge(self.context, logged_in)
+        if DEBUG:
+            print("AddName context:\n{}".format(self.context))
+        self.cloudant_online_store.add_customer_obj(self.customer)
+
+        return True
```

Comment:
Yeah, that pattern is real clunky. It all relates to whether we're doing something in the App that does not require further input from slack, i.e. a DB lookup or adding to the OnlineStoreCustomer, or whether we want to go back to slack for more user input.
I think I can handle all of that in state(context) and it'd be cleaner. For the future...